### PR TITLE
vhosts: give full permissions to default guest administrator

### PR DESCRIPTION
**Pull request by paulrbr-fl**

### CODE REVIEW

File: tasks/rabbitmq_vhosts.yml [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
   run_once: "{{ rabbitmq_enable_clustering is defined and rabbitmq_enable_clustering }}"
   delegate_to: "{{ rabbitmq_master|default(omit) }}"
+  register: rabbitmq_created_vhosts
+
+- name: rabbitmq_extra_vhosts | Check guest administrator is present
+  command: rabbitmqctl -q list_users
+  become: true
+  run_once: "{{ rabbitmq_enable_clustering is defined and rabbitmq_enable_clustering }}"
+  delegate_to: "{{ rabbitmq_master|default(omit) }}"
+  when: rabbitmq_created_vhosts.changed
```
Comment (gaetanfl) [inline tasks/rabbitmq_vhosts.yml:16]: Can you add a changed_when: false to indcate it's read only ?

File: tasks/rabbitmq_vhosts.yml [hunk 2]
```diff
@@ -1,8 +1,27 @@
...
+  register: rabbitmq_existing_users
+
+- name: rabbitmq_extra_vhosts | Give access to new vhosts to guest administrator
+  command: "rabbitmqctl -q set_permissions -p {{ item['name'] }} guest '.*' '.*' '.*'"
+  become: true
+  run_once: "{{ rabbitmq_enable_clustering is defined and rabbitmq_enable_clustering }}"
+  delegate_to: "{{ rabbitmq_master|default(omit) }}"
+  with_items: "{{ rabbitmq_created_vhosts.results|selectattr('changed')|list }}"
+  when:
+    - item['state'] == 'present'
```
Comment (gaetanfl) [inline tasks/rabbitmq_vhosts.yml:27]: Shouldn't there is a and with rabbitmq_created_vhosts.created ?

Comment (paulrbr-fl) [reply tasks/rabbitmq_vhosts.yml:27]: let me check

Comment (gaetanfl) [reply tasks/rabbitmq_vhosts.yml:27]: Maybe the with_items will result in an empty set

Comment (paulrbr-fl) [reply tasks/rabbitmq_vhosts.yml:27]: Unfortunately there is no `created` flag. Cf:
```
    "rabbitmq_created_vhosts": {
        "changed": true,
        "msg": "All items completed",
        "results": [
            {
                "_ansible_delegated_vars": {
                    "ansible_delegated_host": "rabbit_01",
                    "ansible_host": "172.10.0.1"
                },
                "_ansible_ignore_errors": null,
                "_ansible_item_label": {
                    "name": "apps",
                    "state": "present"
                },
                "_ansible_item_result": true,
                "_ansible_no_log": false,
                "_ansible_parsed": true,
                "changed": false,
                "failed": false,
                "invocation": {
                    "module_args": {
                        "name": "apps",
                        "node": "rabbit",
                        "state": "present",
                        "tracing": false
                    }
                },
                "item": {
                    "name": "apps",
                    "state": "present"
                },
                "name": "apps",
                "state": "present"
            },
            {
                "_ansible_delegated_vars": {
                    "ansible_delegated_host": "rabbit_01",
                    "ansible_host": "172.10.0.1"
                },
                "_ansible_ignore_errors": null,
                "_ansible_item_label": {
                    "name": "paulrbr",
                    "state": "present"
                },
                "_ansible_item_result": true,
                "_ansible_no_log": false,
                "_ansible_parsed": true,
                "changed": true,
                "failed": false,
                "invocation": {
                    "module_args": {
                        "name": "paulrbr",
                        "node": "rabbit",
                        "state": "present",
                        "tracing": false
                    }
                },
                "item": {
                    "name": "paulrbr",
                    "state": "present"
                },
                "name": "paulrbr",
                "state": "present"
            }
        ]
    }
```

However checking `item['changed'] && item['state'] == 'present'` represents the `created` flag you expect :)

File: tasks/rabbitmq_vhosts.yml (hunk 1 above)
Comment (paulrbr-fl) [reply tasks/rabbitmq_vhosts.yml:16]: :+1:

**Review by gaetanfl — APPROVED** [review]
Nicely done

**Pull request merged**